drivers/timer/cc23x0_rtc_timer.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2024 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_cc23x0_rtc_timer
+
+#include <soc.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/irq.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/sys/util.h>
+
+#include <inc/hw_rtc.h>
+#include <inc/hw_types.h>
+#include <inc/hw_evtsvt.h>
+#include <inc/hw_memmap.h>
+
+#define RTC_TIMEOUT_MAX 0xFFBFFFFFU
+
+/* Set rtc interrupt to lowest priority */
+#define SYSTIM_ISR_PRIORITY 3U
+
+/* Keep track of rtc counter at previous announcement to the kernel */
+static uint32_t last_rtc_count;
+
+static void rtc_isr(const void *arg);
+static int sys_clock_driver_init(void);
+
+#define TICK_PERIOD_MICRO_SEC (1000000 / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+void sys_clock_set_timeout(int32_t ticks, bool idle)
+{
+	ARG_UNUSED(idle);
+
+	/* If timeout is necessary */
+	if (ticks != K_TICKS_FOREVER) {
+		/* Get current value as early as possible */
+		uint32_t ticks_now = HWREG(RTC_BASE + RTC_O_TIME8U);
+
+		if ((ticks_now + ticks) >= RTC_TIMEOUT_MAX) {
+			/* Reset timer and start from 0 */
+			HWREG(RTC_BASE + RTC_O_CTL) = 0x1;
+			HWREG(RTC_BASE + RTC_O_CH0CC8U) = ticks;
```

Comment:
I did not quite understand this, but could it be, that this will make the calculation of the elapsed tick inaccurate, because the timer was reset and the last_rtc_count was not?